# Práctica Calificada 1 - D'ALESSANDRO POLO GONZALES

**Tópicos de Finanzas Avanzadas (ECON-421, UPAO 2026-20)**

Miércoles 16 de setiembre de 2026. Cubre las semanas 1 a 3. Puntaje total: 20 puntos. Tiempo sugerido: 45 minutos.

## Reglas del juego

Esta es una evaluación con **libro abierto de verdad**: pueden usar internet, sus apuntes, el repositorio del curso, Python y cualquier IA (ChatGPT, Claude, Gemini, la que quieran). No es trampa, es la forma en que van a trabajar como profesionales.

A cambio, las reglas son dos:

1. **Las respuestas escritas van con sus palabras y aplicadas a su caso.** Una respuesta genérica pegada de una IA, correcta pero que podría ser de cualquier alumno y cualquier empresa, obtiene la mitad del puntaje del ítem. Lo que evalúo es que ustedes entiendan, no que su IA entienda.
2. **El trabajo es individual.** Pueden conversar con una IA, no entre ustedes.

## Cómo se entrega

1. Completa este notebook (respuestas escritas en las celdas de texto, código donde se pida).
2. `Kernel  Restart & Run All` para verificar que corre completo.
3. Exporta a PDF: en Colab, menú Archivo, Imprimir, Guardar como PDF; en VS Code o Jupyter, Ctrl+P e imprimir a PDF.
4. Sube el PDF a Canvas antes del fin de la sesión. Como respaldo, haz también commit del notebook en tu fork.

## Parte 1: los conceptos con tus palabras (4 pts, 1 c/u)

Responde cada pregunta en un máximo de dos oraciones, **con tus palabras y usando tu empresa** (la de tus tareas 2 y 3) como ejemplo donde aplique.

**1.1** La semana 1 dijimos que el precio de mercado y el valor intrínseco de una acción pueden diferir. ¿Por qué esa diferencia no te dice, por sí sola, cuánto puedes ganar comprando la acción?

*Tu respuesta:* *El mispricing percibido entre el valor intrínseco y el precio de mercado no representa por sí solo una ganancia, porque nuestra valoración puede contener un error de estimación, ya sea por la información disponible o los supuestos utilizados. Por ejemplo, si estimamos que Coca-Cola vale más de lo que cotiza, parte de ese mispricing podría deberse a un error en nuestra propia valoración.*

**1.2** ¿Por qué la tasa de descuento de tu empresa no es un número que se busca en internet sino un número que se construye? Nombra dos insumos que tuviste que decidir tú al construirla.

*Tu respuesta:* *La tasa de descuento no es un porcentaje hallado en internet, sino que la construimos a partir de los supuestos usados para calcular el Ke por ejemplo. Entre ellos sería el Beta que dependiedo de la empresa tendríamos que buscar similares en el sector, y la prima de riesgo dependiendo qué supuesto usemos para calcularlo. Para mi empresa, KO dependería que enfoque de prima de riesgo estamos usando, si histórico o implícito como lo vimos en clase.*

**1.3** Explica a un gerente sin formación financiera por qué la utilidad neta de su empresa puede crecer mientras la caja disponible cae.

*Tu respuesta:* *Como vimos en la semana 3, la utilidad neta se calcula bajo el principio del devengado, por lo que una empresa puede reconocer mayores ventas y utilidades sin haber recibido aún ese dinero en caja. Por ejemplo, en Coca-Cola (KO), si aumentan sus ventas al crédito o realiza mayor CAPEX, puede presentar una menor caja disponible aunque su utilidad neta haya aumentado.*


**1.4** ¿Cuál es la diferencia entre lo que tu empresa podría repartir a sus accionistas y lo que efectivamente les reparte? ¿Cómo se llama cada concepto?

*Tu respuesta:* *Como vimos en la semana 3, el FCFE representa la caja que nuestra empresa podría repartir a sus accionistas después de cubrir sus operaciones y obligaciones, mientras que los dividendos son lo que efectivamente decide repartir. En el caso de Coca-Cola (KO), aunque genere un FCFE elevado, no necesariamente distribuirá todo como dividendos, ya que puede reinventir en la empresa.*


## Parte 2: encuentra los errores (6 pts)

Un practicante valoró a la empresa Andina SAC y entregó este trabajo. Sus datos de partida son correctos: EBIT = 50, t = 30 por ciento, depreciación = 10, capex = 15, incremento del capital de trabajo = 5, gasto por intereses = 8, endeudamiento neto = 0, Ke = 12 por ciento, WACC = 9 por ciento.

> **Paso 1.** FCFF = EBIT(1-t) + Dep + Int - Capex - WCInv = 35 + 10 + 8 - 15 - 5 = **33**
>
> **Paso 2.** FCFE = FCFF - Int = 33 - 8 = **25**
>
> **Paso 3.** Como la empresa creció 7 por ciento en los últimos años, uso g = 7 por ciento a perpetuidad. Valor del equity = FCFE por (1+g) entre (WACC - g) = 25 por 1.07 entre 0.02 = **1,337.5**

El trabajo contiene **cuatro errores conceptuales**. Identifícalos (2 pts), explica en una oración por qué cada uno es un error (2 pts), y recalcula el valor del equity corrigiendo todo, con un g perpetuo de 3 por ciento (2 pts). Puedes usar la celda de código.

*Errores y explicaciones:*

1. En el paso 1, el EBIT significa la utilidad operativa antes de intereses por lo que adicionar nuevamente los intereses estaría sobreestimando
2. En el paso 2, hace falta descontar el interés después de impuesto Int(1-t) porque la empresa podría utilizar el escudo tributario de tener deuda. 
3. Como lo hablamos en la última clase, en el paso 3, tener un crecimiento de 7% a perpetuidad no es sostenible en el largo plazo
4. En el  último paso el error es que por regla de consistencia el FCFE se descuenta con Ke en lugar de WACC.

In [1]:
# Recalculo corregido (puedes usar utils/finanzas.py si trabajas en tu fork)

# Mi código
EBIT = 50
t = 0.30
dep = 10
capex = 15
wc_inv = 5
intereses = 8
endeudamiento_neto = 0

Ke = 0.12
g = 0.03

FCFF = EBIT * (1 - t) + dep - capex - wc_inv
FCFE = FCFF - intereses * (1 - t) + endeudamiento_neto
valor_equity = FCFE * (1 + g) / (Ke - g)

print(f"FCFF corregido: {FCFF:.2f}")
print(f"FCFE corregido: {FCFE:.2f}")
print(f"Valor del Equity: {valor_equity:.2f}")


FCFF corregido: 25.00
FCFE corregido: 19.40
Valor del Equity: 222.02


*Valor del equity corregido:* **222.02**

## Parte 3: tu empresa (6 pts)

Con la empresa de tus tareas (usa `yfinance` o los resultados que ya tienes de la tarea 2):

**3.1** Calcula el FCFF del último año disponible por la ruta del EBIT. Muestra los insumos que usaste (EBIT, tasa efectiva, depreciación, capex, capital de trabajo). (3 pts)

In [2]:
# TU CODIGO AQUI: FCFF del ultimo anio de tu empresa, ruta EBIT
import yfinance as yf
import pandas as pd

# Empresa
ticker = yf.Ticker("KO")

# Estados financieros anuales
income = ticker.financials
cashflow = ticker.cashflow
balance = ticker.balance_sheet

# Último año disponible
ultimo_anio = income.columns[0]

print("Último año disponible:", ultimo_anio)
print("\nEBIT:")
print(income.loc["EBIT", ultimo_anio])

print("\nDepreciación y amortización:")
print(cashflow.loc["Depreciation And Amortization", ultimo_anio])

print("\nCAPEX:")
print(cashflow.loc["Capital Expenditure", ultimo_anio])

# Tasa efectiva de impuestos

pretax_income = income.loc["Pretax Income", ultimo_anio]
tax_provision = income.loc["Tax Provision", ultimo_anio]

tasa_efectiva = tax_provision / pretax_income

print(f"Utilidad antes de impuestos: ${pretax_income:,.0f}")
print(f"Impuesto: ${tax_provision:,.0f}")
print(f"Tasa efectiva: {tasa_efectiva:.2%}")

# Capital de trabajo

current_assets = balance.loc["Current Assets", ultimo_anio]
current_liabilities = balance.loc["Current Liabilities", ultimo_anio]

anio_anterior = income.columns[1]

current_assets_ant = balance.loc["Current Assets", anio_anterior]
current_liabilities_ant = balance.loc["Current Liabilities", anio_anterior]

capital_trabajo = current_assets - current_liabilities
capital_trabajo_ant = current_assets_ant - current_liabilities_ant

delta_wc = capital_trabajo - capital_trabajo_ant

print(f"Capital de trabajo último año: ${capital_trabajo:,.0f}")
print(f"Capital de trabajo año anterior: ${capital_trabajo_ant:,.0f}")
print(f"Δ Capital de trabajo: ${delta_wc:,.0f}")

# Cálculo del FCFF por la ruta del EBIT

EBIT = income.loc["EBIT", ultimo_anio]
depreciacion = cashflow.loc["Depreciation And Amortization", ultimo_anio]
capex = abs(cashflow.loc["Capital Expenditure", ultimo_anio])

FCFF = EBIT * (1 - tasa_efectiva) + depreciacion - capex - delta_wc

print("----- FCFF DE COCA-COLA -----")
print(f"EBIT: ${EBIT:,.0f}")
print(f"Tasa efectiva: {tasa_efectiva:.2%}")
print(f"Depreciación y amortización: ${depreciacion:,.0f}")
print(f"CAPEX: ${capex:,.0f}")
print(f"Δ Capital de trabajo: ${delta_wc:,.0f}")
print(f"\nFCFF: ${FCFF:,.0f}")

Último año disponible: 2025-12-31 00:00:00

EBIT:
17652000000.0

Depreciación y amortización:
1050000000.0

CAPEX:
-2112000000.0
Utilidad antes de impuestos: $15,998,000,000
Impuesto: $2,861,000,000
Tasa efectiva: 17.88%
Capital de trabajo último año: $9,763,000,000
Capital de trabajo año anterior: $748,000,000
Δ Capital de trabajo: $9,015,000,000
----- FCFF DE COCA-COLA -----
EBIT: $17,652,000,000
Tasa efectiva: 17.88%
Depreciación y amortización: $1,050,000,000
CAPEX: $2,112,000,000
Δ Capital de trabajo: $9,015,000,000

FCFF: $4,418,207,151


**3.2** Para valorar el equity de tu empresa, ¿usarías FCFF o FCFE? Justifica en dos o tres oraciones mirando su situación real: cómo ha venido cambiando su deuda y si su FCFE es positivo. (2 pts)

*Tu respuesta:* *Usaría el FCFE para valorar directamente el equity de Coca-Cola, ya que en 2025 la empresa tuvo un endeudamiento neto positivo y su FCFE se mantiene positivo. Esto permite valorar la caja que queda disponible para sus accionistas considerando cambios es su deuda.*


**3.3** El interés que pagó tu empresa el año pasado, ¿a qué financista fue y por qué NO se resta al calcular el FCFF? (1 pt)

*Tu respuesta:* *El interés que pagó Coca-Cola fue principalmente a sus acreedores financieros, como los tenedores de bonos y otras entidades que le prestaron dinero. No se resta en el FCFF porque este representa la caja disponible para todos los que financian la empresa, tanto acreedores como accionistas, antes del pago de intereses.*


## Parte 4: tú y tu IA (4 pts)

Esta parte evalúa cómo usaste las herramientas, que es una competencia del curso.

**4.1** Pega el prompt más útil que le diste a una IA durante esta evaluación (o durante las tareas, si hoy no la usaste). (1 pt)

*Prompt:* *Actúa como apoyo en esta parte de mi evaluación de Tópico de Finanzas Avanzadas. Ayúdame a calcular el FCFF de Coca-Cola (KO) usando yfinance y la ruta del EBIT. Quiero que me indiques paso a paso qué datos debo obtener de los estados financieros, como EBIT, tasa efectiva de impuestos, depreciación, CAPEX y cambio en capital de trabajo, y que me des el código en Python para calcularlo y mostrar cada insumo.*

**4.2** Señala algo que la IA respondió de forma incompleta, genérica o directamente mal, y cómo lo detectaste o corregiste tú. Si crees que todo lo que te dijo estuvo perfecto, explica cómo lo verificaste. (3 pts)

*Tu respuesta:* *Una respuesta que tuve que corregir fue el recálculo del valor del equity, porque inicialmente se estaban usando un Ke de 10% y un crecimiento perpetuo de 4%, cuando el enunciado indicaba Ke de 12% y g de 3%. Lo detecté revisando los datos del ejercicio y luego verifiqué nuevamente el cálculo en Python, obteniendo un FCFF de 25, un FCFE de 19.40 y un valor del equity de 222.02*


---

**Antes de entregar:** Restart & Run All, exportar a PDF, subir a Canvas. Commit en el fork como respaldo:

```bash
git add examenes/clase03_pc1.ipynb
git commit -m "PC1"
git push
```